sdks/java/extensions/smb/src/main/java/org/apache/beam/sdk/extensions/smb/SMBFilenamePolicy.java

```diff
@@ -81,14 +88,16 @@ Long getTempId() {
     private final ResourceId filenamePrefix;
     private final String filenameSuffix;
     private final boolean doTimestampFiles;
+    private final Compression compression;
 
     FileAssignment(ResourceId filenamePrefix, String filenameSuffix, boolean doTimestampFiles) {
       this.filenamePrefix = filenamePrefix;
       this.filenameSuffix = filenameSuffix;
       this.doTimestampFiles = doTimestampFiles;
+      this.compression = Compression.detect(filenameSuffix);
```

Comment:
We should allow overriding `compression`, since it may not always be detectable via suffix?